### PR TITLE
feat(react): custom properties are now sortable by name in the UI (#2…

### PR DESCRIPTION
**Pull request by RickardCardell**
## Checklist
- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [X] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by RickardCardell** [comment]
Small demo:

image

**Review by gabe-lyons — APPROVED** [review]
looks good to me! thanks @RickardCardell

File: datahub-web-react/src/app/entity/shared/Properties.tsx [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 };
 
 export function Properties({ properties }: Props) {
-    const propertyTableColumns = [
+    const propertyTableColumns: ColumnsType<StringMapEntry> = [
         {
             title: 'Name',
             dataIndex: 'key',
+            sorter: (a, b) => b.key.localeCompare(a.key),
+            defaultSortOrder: 'descend',
```
Comment (shirshanka) [inline datahub-web-react/src/app/entity/shared/Properties.tsx:16]: @RickardCardell : is the intent to have the default display order be a-z? or z-a?

File: datahub-web-react/src/app/entity/shared/Properties.tsx [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
     properties: StringMapEntry[];
 };
 
 export function Properties({ properties }: Props) {
-    const propertyTableColumns = [
+    const propertyTableColumns: ColumnsType<StringMapEntry> = [
         {
             title: 'Name',
             dataIndex: 'key',
+            sorter: (a, b) => b.key.localeCompare(a.key),
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/Properties.tsx:15]: should this be `a.key.localeCompare(b.key)`?

File: datahub-web-react/src/app/entity/shared/Properties.tsx (hunk 1 above)
Comment (RickardCardell) [reply datahub-web-react/src/app/entity/shared/Properties.tsx]: Nice catch. Intention is 'a-z'. I've pushed a fix

File: datahub-web-react/src/app/entity/shared/Properties.tsx (hunk 2 above)
Comment (RickardCardell) [reply datahub-web-react/src/app/entity/shared/Properties.tsx]: Yes, indeed. Nice catch. The combined `b.key.localeCompare(a.key)` and `descend` resulted in ascending order. I've pushed a fix

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**